### PR TITLE
[MRG] Update revcomp from khmer

### LINKED ISSUE #330: add faster revcomp in from khmer
https://github.com/dib-lab/khmer/pull/1791                                      

via @betatim

### PR DESCRIPTION
**Pull request by luizirber**
Fixes #330 (stolen from https://github.com/dib-lab/khmer/pull/1791 )

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

File: sourmash_lib/kmer_min_hash.hh [hunk 1]
```diff
@@ -190,32 +195,17 @@ public:
...
-                complement = 'G';
-                break;
-            case 'G':
-                complement = 'C';
-                break;
-            case 'T':
-                complement = 'A';
-                break;
-            default:
-                std::string msg = "invalid DNA character in sequence: ";
```
Comment (luizirber) [inline sourmash_lib/kmer_min_hash.hh]: We don't have any tests that trigger this behavior (and this is different than what `khmer` does)

Comment (ctb) [reply sourmash_lib/kmer_min_hash.hh]: I am confused. We certainly _revcomp a lot....?

Comment (luizirber) [reply sourmash_lib/kmer_min_hash.hh]: But `_revcomp` is not exposed to Python, so it was never tested. But I think we can trigger the behavior in `add_hash`, I'm gonna try.

**Comment by ctb** [comment]
I'm still confused as it should be run for every add_sequence.  Is this
duplicate code or something?

File: sourmash_lib/kmer_min_hash.hh (hunk 1 above)
Comment (luizirber) [reply sourmash_lib/kmer_min_hash.hh]: No, we can't: `_checkdna` fails before we can use `_revcomp` with an invalid base.

**Comment by ctb** [comment]
oh, you're just talking about the non-ACGT characters. Yes.

**Comment by luizirber** [comment]
Ready for review and merge

**Comment by ctb** [comment]
This branch:
```
% time python ./sourmash compute -k 31 --scaled 10000 reads1.fa -f
...
calculated 1 signatures for 1000000 sequences in reads1.fa

real    0m28.773s
user    0m28.117s
sys     0m0.283s
```

Current master:
```
time python ./sourmash compute -k 31 --scaled 10000 reads1.fa -f
...
calculated 1 signatures for 1000000 sequences in reads1.fa

real    0m34.251s
user    0m33.597s
sys     0m0.268s
```

So about 10-15% faster. Cool!

**Pull request merged**